### PR TITLE
Add a repeating texture example

### LINKED ISSUE #11136: Add an example demonstrating how to repeat textures
This is a simple adoption of https://github.com/bevyengine/bevy/issues/11109.

### PR DESCRIPTION
**Pull request by IQuick143**
# Objective

- Make an example on how to set custom properties on samplers and achieve a repeating texture.
- Fixes #11136
- Progress towards #399 

## Changelog

Started with the commits from the PR #11109 
Added another commit that
- Re-orders functions (main -> setup -> other)
- Generalises the function `quad_1_2` to `quad_with_custom_uv` to make the sample code more useful and reusable
- Added a comment explaining why we are manipulating UVs and how texture sampler tie into that

## Further details

- Should we build the quad mesh from scratch, or show how to change the UVs on a mesh created by `bevy::render::mesh::shape::Quad`?
- Probably wait on #11111 being fixed to resolve the need for a duplicate texture.
- Could/should the code for setting the sampler properties be written nicer (lines setting the actual properties are effectively duplicated for the two cases)?
- Any further suggestions?

### CODE REVIEW

**Comment by IQuick143** [comment]
@alice-i-cecile @IceSentry would you mind reviewing this?

File: examples/asset/texture_sampler.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                sampler.address_mode_v = ImageAddressMode::Repeat;
+            }
+        },
+    );
+
+    // Instead of using a standard quad with UV coordinates in the range `0..=1` (spanning the entire texture),
+    // we create a quad with UV coordinates in the range `-1..=2`, going beyond the texture by one entire unit in each direction.
+    // Texture behaviour then depends on the selected sampler mode.
+    // By default (`ImageAddressMode::ClampToEdge`) the out-of-bounds UV coordinates will map to the edge of the texture.
+    // When `ImageAddressMode::Repeat` is specified the out-of-bounds UV coordinates will repeat the texture.
```
Comment (stepancheg) [inline examples/asset/texture_sampler.rs:76]: My opinion, which has very little weight in this context would be this:

Less text is better, because those who read the example, would have to spend less time and energy reading it.

> standard quad with UV coordinates

There's no such thing as "standard quad with UV coordinates". For example, if you have a mesh for a sphere, it usually contains hundreds quads, and none of them use coordinates 0..1. Quads with XY and UV coordinates 0..1 are used mostly in examples (and probably in Minecraft).

Also the comments below "usual UV range `0..=1`", "custom UV coordinates" are not exactly correct for the same reason.

> Texture behaviour then depends on the selected sampler mode

Texture behavior always depends on the sampler more, not "then". Perhaps you meant "Texture behaviour then depends on the selected **address** mode", which indeed *then* depends, because when UV is in range 0..=1, it does not depend on address mode.

> By default (`ImageAddressMode::ClampToEdge`) the out-of-bounds UV coordinates will map to the edge of the texture.
> When `ImageAddressMode::Repeat` is specified the out-of-bounds UV coordinates will repeat the texture

These would be the redundant comments because they
- repeat the mostly obvious enum variant name
- repeat enum documentation
- basically repeat what is said several lines above where the image is loaded (where this comments belongs)

Again, feel free to ignore this comment.

**Comment by stepancheg** [comment]
> Generalises the function quad_1_2 to quad_with_custom_uv to make the sample code more useful and reusable

In my very subjective opinion, examples are not meant to optimized for reading, not for code reuse.

File: examples/asset/texture_sampler.rs (hunk 1 above)
Comment (IQuick143) [reply examples/asset/texture_sampler.rs:76]: You make a good point in regards to the **sampler** vs **address** mode, that should be reworded.

As per what a standard quad UV means, you're right that when you're talking about quad as a graphics primitive (part of a mesh), it could have any UVs for any reason, but when one talks about a quad as a mesh itself (a geometric quadrilateral) it does occur to me only as natural to use UV coordinates as usual.
I said usual UV range `0..=1`, because in that range lie UVs of most meshes people come into contact with (3D designers don't typically make their UVs sample outside of a texture from my experience).

You're right the comment does seem to give a bunch of redundant information that could be found in the docstrings for `ImageAddressMode`, though the purpose of the comment is mainly to explain why it takes user-handled in UV coordinates + a change in address mode to achieve the desired repeating effect, because just changing the adress mode would not actually produce any noticeable effects.
I'll consider shortening this part to a refference to the appropriate docs.

Anyway if someone gives me another opinion on this, I'll rewrite it.

**Comment by rparrett** [comment]
Could you edit your PR description to include that this also fixes #399?

Also, I think we can simplify and unblock this by just removing the non-repeating texture from the example. Discussion: https://github.com/bevyengine/bevy/issues/11136#issuecomment-1888458231

**Comment by IQuick143** [comment]
> Could you edit your PR description to include that this also fixes #399?
> 
> Also, I think we can simplify and unblock this by just removing the non-repeating texture from the example. Discussion: [#11136 (comment)](https://github.com/bevyengine/bevy/issues/11136#issuecomment-1888458231)

Issue #399 added!
I think the example is better when it shows directly the effects of different `ImageAdressMode`s, but that's maybe just my opinion. The argument that it would unblock this is somewhat compelling but I think that since #11111 needs to be fixed anyway, this can probably wait (perhaps reconsider if this block would decide if this PR makes it into 0.13 or not).

File: examples/asset/texture_sampler.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        ..default()
+    });
+}
+
+/// Creates a quad with UV coordinates in range `uv_low..=uv_high`, instead of the usual UV range `0..=1`.
+fn quad_with_custom_uv(uv_low: f32, uv_high: f32) -> Mesh {
+    let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
+    mesh.insert_attribute(
+        Mesh::ATTRIBUTE_POSITION,
+        vec![[0., 0., 0.], [1., 0., 0.], [1., 1., 0.], [0., 1., 0.]],
```
[the remark this answers is not in the document]
Comment (IQuick143) [reply examples/asset/texture_sampler.rs:107]: Modifying the default Quad might be better, I'll at least fix the mesh coordinates.

A better function name is welcome, although `repeating_quad(times: f32)` feels a little suggestive, because it suggest using this mesh would automatically cause repeating, whereas it only sets up a mesh that can be repeated over, but maybe this is not such a problem given the comments explaining the mechanisms.

**Comment by bugsweeper** [comment]
> Please correct me if I'm wrong.

StandardMaterial also has uv_transform since #11904 , which is done exactly for GLTF

**Comment by janhohenheim** [comment]
@bugsweeper yep, I wrote that PR 😉
